allow use of sk_sg_filter by passing channel bandwidth and sampling interval

mef51 opened the pull request:
`sk_sg_filter` only uses `foff` and `tsamp` from the `Your` class and I wanted to apply this on waterfalls without having to instantiate this class to use it. This change allows either use case: use with `your_object` or optionally pass `foff` and `tsamp` instead

devanshkv commented:
@mef51 thanks for the PR, this looks like a good idea! Can you please run [black](https://github.com/psf/black) on the your code?

mef51 commented:
@devanshkv sure, done

devanshkv commented on your/utils/rfi.py:
```diff
@@ -131,6 +133,8 @@ def sk_sg_filter(
     Args:
         data (numpy.ndarray): 2D frequency time data
         your_object: Your object
+        foff (float): Channel bandwidth (MHz)
```
it would be good to mention in the documentation that if both `your_object` and `foff` and/or `tsamp` is provided, the code will use the user-provided `foff` and/or `tsamp` values.

mef51 replied:
https://github.com/thepetabyteproject/your/pull/83/files#diff-95754a8efb570ba353c90b74c1acf3b080c9ffa51e9c6a0f44e9825c8e310002R139

devanshkv commented:
@mef51 can you please modify the following test to include cases for your code:
https://github.com/thepetabyteproject/your/blob/3d286d9d679c9e26dc27eab4fa2c5088dd7573cf/tests/test_rfi.py#L46-L56

mef51 commented:
The test was failing because of the different signature. I moved the new kwargs to the end of the arg list to preserve the signature. Also added a test case for the new kwargs

devanshkv commented:
Awesome! Thank you so much! We are going to make a release soon and will make sure to detail your contributions and tag you.

mef51 commented:
Thanks! Appreciate it.
